### PR TITLE
Requirements upgrades

### PR DESCRIPTION
**Pull request by deinok**
Fixes old requirements - Only the requirements that don't break anything
Sort requirements.in
Adds a badge to README that shows the actual state of the requirements

### CODE REVIEW

File: requirements.in [hunk 1]
```diff
@@ -1,35 +1,35 @@
...
 fanstatic==0.12
-Markdown==2.4
-ofs==0.4.1
+Flask==0.10.1
+Jinja2==2.8
+Markdown==2.6.6
+ofs==0.4.2
 ordereddict==1.1
-Pairtree==0.7.1-T
+Pairtree==0.5.8
```
Comment (amercader) [inline requirements.in]: This looks like a downgrade

File: README.rst [hunk 2]
```diff
@@ -20,6 +20,10 @@ CKAN: The Open Source Data Portal Software
     :target: https://coveralls.io/github/ckan/ckan?branch=master
     :alt: Coverage Status
 
+.. image:: https://requires.io/github/ckan/ckan/requirements.svg?branch=master
```
Comment (amercader) [inline README.rst]: Maybe when we get rid of Pylons we can have a badge to brag about up to date requirements, until then it will always show "outdated" which doesn't look too good

File: requirements.in (hunk 1 above)
Comment (deinok) [inline requirements.in]: True, Requires.io marks 0.5.8 as the latest, the "T" confused me, I think that it was a Beta release or something. I send comit with the two fixes

**Comment by deinok** [comment]
@amercader Done. What about the usage of Bower for Javascript and CSS Vendor Librarys?

File: requirements.in [hunk 3]
```diff
@@ -1,35 +1,35 @@
 # The file contains the direct ckan requirements.
 # Use pip-compile to create a requirements.txt file from this
 Babel==2.3.4
-Jinja2==2.8
-Pylons==0.9.7
 Beaker==1.7.0 # need to pin this because of https://github.com/bbangert/beaker/commit/fc511ceda4305fcf54919b3f081fed7790e2fef3
```
Comment (amercader) [inline requirements.in]: Beaker should be also safe to upgrade.

The comment should be something like `# Needs to be pinned to a more up to date version than the Pylons one`

File: requirements.txt [hunk 4]
```diff
@@ -7,54 +7,54 @@
...
+pyutilib.component.core==4.6.4
 repoze.lru==0.6           # via routes
 repoze.who-friendlyform==1.0.8
-repoze.who==2.0
-requests==2.7.0
+repoze.who==2.3
+requests==2.10.0
 Routes==1.13
-simplejson==3.3.1         # via pylons HAND-FIXED FOR NOW #2681
+simplejson==3.8.2         # via pylons HAND-FIXED FOR NOW #2681
```
Comment (amercader) [inline requirements.txt]: LazyJSONObject has been removed in #2681 so we can upgrade delete this comment

File: requirements.in [hunk 5]
```diff
@@ -1,35 +1,35 @@
...
+pyutilib.component.core==4.6.4
 repoze.who-friendlyform==1.0.8
-repoze.who==2.0
-requests==2.7.0
+repoze.who==2.3
+requests==2.10.0
 Routes==1.13
-sqlalchemy-migrate==0.9.1
+sqlalchemy-migrate==0.10.0
 SQLAlchemy==0.9.6
```
Comment (amercader) [inline requirements.in]: Did you left out SQLAlchemy for reason in particular? It's a critical one so it can probably go into a separate PR

**Comment by amercader** [comment]
@deinok All looks good apart from the minor comments I added

Bower for frontend sounds good but that's a separate issue

File: requirements.in (hunk 5 above)
Comment (deinok) [inline requirements.in]: I only left the libs that make test to fail, they will be upgraded on separated PR for each one, so we can merge quickly

File: requirements.in (hunk 3 above)
Comment (deinok) [inline requirements.in]: Done

File: requirements.txt (hunk 4 above)
Comment (deinok) [inline requirements.txt]: Done

**Comment by deinok** [comment]
@amercader Changes done ;)

**Comment by amercader** [comment]
Thanks @deinok !

**Comment by amercader** [comment]
@ckan/core time to upgrade your local requirements on master!
